test/integration/sidecar/sidecar_stack_test.go

```diff
@@ -0,0 +1,50 @@
+package sidecar
+
+import (
+	"context"
+
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("sidecar features", func() {
+	var ctx context.Context
+	var f *framework.Framework
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		f = framework.New(framework.GlobalOptions)
+	})
+
+	Context("wait for sidecar to initialize", func() {
+		var stack *SidecarStack
+		var err error
+
+		BeforeEach(func() {
+			stack, err = newSidecarStack("sidecar-test", framework.GlobalOptions.KubeConfig)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			stack.cleanup(ctx, f)
+		})
+
+		It("should have the color annotation", func() {
+			stack.createMeshAndNamespace(ctx, f)
+			stack.createBackendResources(ctx, f)
```

Comment:
im waiting until backend pods are bound to a node to ensure that the test does not fail bc the backend is not ready. then the frontend is provisioned and immediately tested for the annotation after its pods are bound to a node